packages/test/src/test-schedules.ts

```diff
@@ -0,0 +1,610 @@
+import { RUN_INTEGRATION_TESTS } from './helpers';
+import anyTest, { TestInterface } from 'ava';
+import { randomUUID } from 'crypto';
+import { Client, Connection, defaultPayloadConverter } from '@temporalio/client';
+import asyncRetry from 'async-retry';
+import { msToNumber } from '@temporalio/common/lib/time';
+import { CalendarSpec, CalendarSpecDescription, ScheduleSummary } from '@temporalio/client/lib/schedule-types';
+import { ScheduleHandle } from '@temporalio/client/lib/schedule-client';
+import { TestWorkflowEnvironment } from '@temporalio/testing';
+import { temporal } from '@temporalio/proto';
+import * as grpc from '@grpc/grpc-js';
+
+export interface Context {
+  testEnv: TestWorkflowEnvironment;
+  client: Client;
+}
+
+const taskQueue = 'async-activity-completion';
+const test = anyTest as TestInterface<Context>;
+
+const dummyWorkflow = async () => undefined;
+const dummyWorkflow2 = async (_x?: string) => undefined;
+
+const calendarSpecDescriptionDefaults: CalendarSpecDescription = {
+  second: [{ start: 0, end: 0, step: 1 }],
+  minute: [{ start: 0, end: 0, step: 1 }],
+  hour: [{ start: 0, end: 0, step: 1 }],
+  dayOfMonth: [{ start: 1, end: 31, step: 1 }],
+  month: [{ start: 'JANUARY', end: 'DECEMBER', step: 1 }],
+  dayOfWeek: [{ start: 'SUNDAY', end: 'SATURDAY', step: 1 }],
+  year: [],
+  comment: '',
+};
+
+test.before(async (t) => {
+  const testEnv = await TestWorkflowEnvironment.createLocal();
+  t.context = {
+    testEnv,
+    client: testEnv.client,
+  };
+  // In case we're running with temporalite or other non default server.
+  // NOTE: at the time this was added temporalite did not expose the grpc OperatorService.
```

Comment:
It actually exposes it and if you start it with `system.forceSearchAttributesCacheRefreshOnRead=true` it works pretty well for testing.